### ISSUE TITLE
Support using Singularity to run builds in the container image

### ISSUE THREAD

**Issue by tsibley** [nextstrain/cli#2]
`nextstrain build` should be able to support running builds using the container image but in an environment which doesn't support docker or have it installed, such as the Hutch's rhino cluster.  This can be accomplished using [Singularity](https://singularity.lbl.gov).  The interface might look like one of these:

```
nextstrain build --singularity .
```

Implementing this would involve adding a new runner, nextstrain/cli/runner/singularity.py, and registering it within the runner framework.

**Comment by pvanheus** [comment]
Hi there 

I'll look into adding it to the wrapper, but if you want to use Singularity with nextstrain you can:

1) build a singularity image (I used a specific tagged version for this, see [Docker Hub](https://hub.docker.com/r/nextstrain/base/tags) for list of tags:

```
singularity build nextstrain-base-build-20200309T035736Z.simg 'docker://nextstrain/base:build-20200309T035736Z'
```
2) running the `ncov` pipeline from within a cloned copy of the [ncov](https://github.com/nextstrain/ncov) github repository, this does `nextstrain build .`

```
singularity exec -B $(pwd):/nextstrain/build nextstrain-base-build-20200309T035736Z.simg  snakemake --printshellcmds
```
For reference, the Docker command that this `nextstrain build .` runs is:

```
docker run --rm --tty --interactive --user=$(id -u):$(id -g) --volume=$(pwd):/nextstrain/build --workdir=/nextstrain/build --env=AUGUR_RECURSION_LIMIT --env=AUGUR_MINIFY_JSON --env=AWS_ACCESS_KEY_ID --env=AWS_SECRET_ACCESS_KEY --env=AWS_SESSION_TOKEN --env=ID3C_URL --env=ID3C_USERNAME --env=ID3C_PASSWORD --env=RETHINK_HOST --env=RETHINK_AUTH_KEY nextstrain/base snakemake --printshellcmds
```

**Comment by tsibley** [comment]
Thanks for looking into this @pvanheus! If you have any questions about how the CLI works, feel free to ask them here.

**Comment by pvanheus** [comment]
This works fairly similarly for ncov-ingest:

```bash
SINGULARITYENV_GISAID_API_ENDPOINT=https://www.epicov.org/BLAH
SINGULARITYENV_GISAID_USERNAME_AND_PASSWORD=USER:PASSWORD
singularity exec /tools/containers/nextstrain/ncov-ingest.sif snakemake --configfile config/local_gisaid.yaml
```

**Comment by tsibley** [comment]
@pvanheus FWIW, the [just-released Nextstrain CLI 6.1.0](https://github.com/nextstrain/cli/releases/tag/6.1.0) now [supports Singularity directly](https://docs.nextstrain.org/projects/cli/en/stable/installation/#singularity). If you want to give it a go, please do report back any issues, suggestions, etc. :-)
